api/scpca_portal/management/commands/load_data.py

```diff
@@ -111,53 +186,74 @@ def load_data(
+            metadata_project_id = project_metadata["scpca_project_id"]
+            passed_project_id = kwargs.get("scpca_project_id")
 
-            if not self.project_has_s3_files(scpca_project_id):
-                logger.warning(
-                    f"Metadata found for '{scpca_project_id}' but no s3 folder of that name exists."
-                )
-                return
-
-            if project_data["pi_name"] not in allowed_submitters:
-                logger.warning("Project submitter is not the white list.")
+            pi_name = project_metadata["pi_name"]
+            if self.skip_project(
+                metadata_project_id, passed_project_id, pi_name, allowed_submitters
+            ):
                 continue
 
-            if project := Project.objects.filter(scpca_id=scpca_project_id).first():
-                # Purge existing projects so they can be re-added.
-                if kwargs["reload_all"] or kwargs["reload_existing"]:
-                    logger.info(f"Purging '{project}")
-                    project.purge(delete_from_s3=kwargs["update_s3"])
-                # Only import new projects.
-                # If old ones are desired they should be purged and re-added.
-                else:
-                    logger.info(f"'{project}' already exists. Use --reload-existing to re-import.")
+            if project := Project.objects.filter(scpca_id=metadata_project_id).first():
+                get_purge_project_kwargs = itemgetter("reload_all", "reload_existing", "update_s3")
+                # If there is a problem purging the project, then skip it
+                if not self.purge_project(project, *get_purge_project_kwargs(kwargs)):
                     continue
 
-            logger.info(f"Importing '{project}' data")
-            project = Project.get_from_dict(project_data)
+            logger.info(f"Importing 'Project {metadata_project_id}' data")
+            project = Project.get_from_dict(project_metadata)
             project.save()
-            Contact.bulk_create_from_project_data(project_data, project)
-            ExternalAccession.bulk_create_from_project_data(project_data, project)
-            Publication.bulk_create_from_project_data(project_data, project)
 
-            project.load_data(**kwargs)
+            Contact.bulk_create_from_project_data(project_metadata, project)
+            ExternalAccession.bulk_create_from_project_data(project_metadata, project)
+            Publication.bulk_create_from_project_data(project_metadata, project)
+
+            project.load_metadata()
             if samples_count := project.samples.count():
                 logger.info(
                     f"Created {samples_count} sample{pluralize(samples_count)} for '{project}'"
                 )
 
+            # Prep callback function
+            partial_kwargs = {
+                k: v for k, v in kwargs.items() if k in ["update_s3", "clean_up_output_data"]
+            }
```

Comment:
Adjust `utils.filter_dict_list_by_keys` to handle this use case.
Also, investigate if we still use `utils.filter_dict_list_by_keys` as it was used before (there have been changes to handling of metadata since that function was written).